source/server/configuration_impl.h

```diff
@@ -44,6 +45,19 @@ class NetworkFilterConfigFactory {
 };
 
 /**
+ * Implemented by each network filter and registered via registerHttpTracerFactory() or
+ * the convenience class RegisterHttpTracerFactory.
+ */
+class HttpTracerFactory {
```

Comment:
It creates a build dependency cycle because //include/envoy/tracing:http_tracer_interface and //include/envoy/server:instance_interface would both depend on one another.